### ISSUE TITLE
mlpack_export.h: No such file or directory

### ISSUE THREAD

**Issue by keonkim** [mlpack/mlpack#633]
mlpack (the newest version in the github) is successfully installed and and are located in `/usr/local/nclude/mlpack` with path configured. Command line applications work perfectly, but I am having a trouble importing it in other programs.

I am getting the following error when I run the simple timer example on my new Ubuntu 16.04 machine:
```
keon@buntu:~/projects/test$ g++ -std=c++11 timer.cpp
In file included from /usr/local/include/mlpack/core.hpp:199:0,
                 from timer.cpp:1:
/usr/local/include/mlpack/core/util/log.hpp:11:27: fatal error: mlpack_export.h: No such file or directory
```
I tried with other simple examples and not being able to import `mlpack_export.h` seems to be the problem because when I manually commented line in log.hpp:11:27 out, same error occurred in other file when it tried to import `mlpack_export.h`.

`mlpack_export.h` exists under `/usr/local/include/mlpack/`, 
so I tried manually adding mlpack_export.h to the PATH but it didn't seem to work.

**Comment by Kirizaki** [comment]
Try to link against mlpack and I'm not sure about order of cmds.

Try like this:

g++ timer.cpp -lmlpack -std=c++11
30 kwi 2016 22:23 "Keon Kim" <notifications@github.com> napisał(a):

**Comment by Kirizaki** [comment]
Ohh, and maybe change Your .cpp File to some different then timer.cpp,
maybe mytimer.cpp.

This shouldn't make any conflict, but I'm away of my pc do I'm guessing a
little.
1 maj 2016 11:24 "Grzegorz Krajewski" <krajekg@gmail.com> napisał(a):

**Comment by zoq** [comment]
Nice catch, I think we should integrate the tutorials code into the build process to avoid such a situation.

So, I think what happens here is that once mlpack is build, the``` mlpack_export.h``` isn't in the right location. Right now I'm not completely sure this is the right way to solve the issue, however here is a simple fix that should work:

Add the following to the ```src/mlpack/CMakeLists.txt``` after line 33.

```
FILE(COPY
    ${CMAKE_CURRENT_BINARY_DIR}/mlpack_export.h
    DESTINATION
    ${CMAKE_SOURCE_DIR}/src/mlpack/)
```

Change ```#include <mlpack_export.h>``` to ```#include <mlpack/mlpack_export.h>``` in all files that include the header (https://github.com/mlpack/mlpack/search?utf8=%E2%9C%93&q=mlpack_export.h&type=Code).

Maybe someone else has a better solution to solve the issue?

**Comment by keonkim** [comment]
I modified the code as @zoq suggested and added `-lmlpack` flag in the command like @Kirizaki  suggested.
It works like a charm now :) thank you.

Should I update the `CMakeLists.txt` and likes changed to `#include <mlpack/mlpack_export.h>` to my future pull request? or is it just my case?

**Comment by rcurtin** [comment]
I think that we should leave this open until it's fixed, since it's a pretty big issue if you can't use mlpack after installation.  I took a look and I think Marcus's comments are right: the `#include <mlpack_export.h>` calls should all be changed to `#include <mlpack/mlpack_export.h>`.  But there's one more piece too:

In `src/mlpack/CMakeLists.txt`, there is a line that adds an include directory for mlpack_export.h (I think it's very near the top); if we change all the #includes, then this line is no longer necessary, so we should remove that too.

@xantares may have more comments on this, since he wrote the `mlpack_export.h` code (in #622).

**Comment by rcurtin** [comment]
I tested @xantares' fix in #634, it seems to work.  In d502539 I moved `mlpack_export.h` to `mlpack_export.hpp` too, just to be more in line with the names of other files in mlpack.  Everything is tested and appears to work.
